test/integration/progressUsers.test.js

```diff
@@ -0,0 +1,187 @@
+        .request(app)
+        .post(`progresses`)
+        .set("Cookie", `${cookieName}=${testUserJwt}`)
+        .send(standupProgressDay2);
+
+      expect(responseFirstNextDayEntry).to.have.status(400);
+      expect(responseFirstNextDayEntry.body).to.have.keys(["message", "error"]);
+      expect(responseFirstNextDayEntry.body.error).to.be.equal("Standup entry already exists for the day");
+      expect(responseFirstNextDayEntry.body.message).to.be.equal("You can only add one standup entry per day");
+
+      // entry for the next day after 6 am
+      clock.setSystemTime(new Date(2022, 5, 3, 6, 30).getTime());
+
+      const responseSecondNextDayEntry = await chai
+        .request(app)
+        .post(`progresses`)
+        .set("Cookie", `${cookieName}=${testUserJwt}`)
+        .send(standupProgressDay2);
+      expect(responseSecondNextDayEntry).to.have.status(200);
+      expect(responseSecondNextDayEntry.body).to.have.keys(["message", "data"]);
+      expect(responseSecondNextDayEntry.body.data).to.have.keys([
+        "id",
+        "userId",
+        "type",
+        "completed",
+        "planned",
+        "blockers",
+        "createdAt",
+        "date",
+      ]);
+      expect(responseSecondNextDayEntry.body.message).to.be.equal("Standup progress created successfully");
+      expect(responseSecondNextDayEntry.body.data.userId).to.be.equal(testUserId);
+    });
+
+    it("Gives 400 for invalid request body", function (done) {
+      const requests = incompleteProgress.map((progress) => {
+        return chai
+          .request(app)
+          .post(`progresses`)
+          .set("Cookie", `${cookieName}=${testUserJwt}`)
+          .send(progress.payload)
+          .then((res) => {
+            expect(res).to.have.status(400);
+            expect(res.body.message).to.include(`Required field '${progress.missingField}' is missing`);
+          });
+      });
+
+      Promise.all(requests)
+        .then(() => done())
+        .catch((err) => done(err));
+    });
+
+    it("Gives 401 for unauthenticated user", function (done) {
+      chai
+        .request(app)
+        .post(`progresses`)
+        .send(standupProgressDay1)
+        .end((err, res) => {
+          if (err) return done(err);
+          expect(res).to.have.status(401);
+          expect(res.body.message).to.be.equal("Access denied: User authentication is required");
+          return done();
+        });
+    });
+  });
+
+  describe("Test the GET Call", function () {
+    let testUserId;
+
+    beforeEach(async function () {
+      testUserId = await addUser(userData[1]);
+      const progressData = stubbedModelProgressData(testUserId, Date.now(), Date.now());
+      await firestore.collection("progresses").doc("testProgressDocument").set(progressData);
+    });
+
+    it("Returns the progress array for the user", function (done) {
+      chai
+        .request(app)
+        .get(`/progresses?userId=${testUserId}`)
```

Comment:
won't we be using type as a user for the standup feature?

tomorrow if my task id and user id clash then we might get a discrepancy

Reply:
are we keeping userId and taskId